### PR TITLE
Implemented: Logic to fetch and populate product list on Find product Inventory Page (#232yz5h).

### PR DESCRIPTION
**Pull request by meet-aniket**
image

### CODE REVIEW

File: src/store/modules/stock/actions.ts [hunk 1]
```diff
@@ -24,20 +24,25 @@ const actions: ActionTree<StockState, RootState> = {
...
-      "fieldsToSelect": ["productId","atp"],
-      "viewSize": productIds.length
-    });
-    if (resp.status === 200 && !hasError(resp)) {
-      // Handled empty response in case of failed query
-      if (resp.data) commit(types.STOCK_ADD_PRODUCTS, { products: resp.data.docs })
+
+    const count = productIds.length / 100;
+
+    for(let i = 0; i < count; i++) {
```
Comment (ymaheshwari1) [inline src/store/modules/stock/actions.ts]: Add a comment why we have done this.

File: src/views/FindProductInventory.vue [hunk 2]
```diff
@@ -139,66 +139,66 @@
...
+                      <Image :src="product.mainImage" />
                     </ion-thumbnail>
                     <ion-label>
-                      <p>Brand</p>
-                      Virtual Name
-                      <p>{{ $t("Color") }} : color</p>
-                      <p>{{ $t("Size") }}: size</p>
+                      <p>{{ product.brand }}</p>
+                      {{ product.productName }}
+                      <p>{{ $t("Color") }} : {{ $filters.getFeature(getProduct(product.productId).featureHierarchy, '1/COLOR/') }}</p>
```
Comment (ymaheshwari1) [inline src/views/FindProductInventory.vue]:
```suggestion
                      <p>{{ $t("Color") }}: {{ $filters.getFeature(getProduct(product.productId).featureHierarchy, '1/COLOR/') }}</p>
```

File: src/views/FindProductInventory.vue [hunk 3]
```diff
@@ -139,66 +139,66 @@
...
-                  <div v-for="item in 2" :key="item" class="list-item">
+                  <div v-for="item in product.variants" :key="item.productId" class="list-item">
                     <div>
                       <ion-item lines="none">
                         <ion-label>
-                          SKU
-                          <p>{{ $t("Color") }} : color</p>
-                          <p>{{ $t("Size") }}: size</p>
+                          {{ item.sku }}
+                          <p>{{ $t("Color") }} : {{ $filters.getFeature(getProduct(item.productId).featureHierarchy, '1/COLOR/') }}</p>
```
Comment (ymaheshwari1) [inline src/views/FindProductInventory.vue]:
```suggestion
                          <p>{{ $t("Color") }}: {{ $filters.getFeature(getProduct(item.productId).featureHierarchy, '1/COLOR/') }}</p>
```

File: src/store/modules/product/actions.ts [hunk 4]
```diff
@@ -83,6 +83,51 @@ const actions: ActionTree<ProductState, RootState> = {
       this.dispatch('product/fetchProducts', { productIds })
       this.dispatch('stock/addProducts', { productIds })
     }
+  },
+
+    /**
+   * Get Product Inventory
+   */
+  async getProductInventory({ commit, state }, payload) {
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts]: This action is not returning product inventory, it's just returning the products information so we can improve it's name to something like fetchProducts

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/stock/actions.ts (hunk 1 above)
Comment (meet-aniket) [reply src/store/modules/stock/actions.ts]: Added a comment in the code.

File: src/views/FindProductInventory.vue (hunk 2 above)
Comment (meet-aniket) [reply src/views/FindProductInventory.vue]: Applied changes.

File: src/views/FindProductInventory.vue (hunk 3 above)
Comment (meet-aniket) [reply src/views/FindProductInventory.vue]: Applied changes.

File: src/store/modules/product/actions.ts (hunk 4 above)
Comment (meet-aniket) [reply src/store/modules/product/actions.ts]: Changed name of the function.

File: src/store/modules/product/actions.ts [hunk 5]
```diff
@@ -83,6 +83,54 @@ const actions: ActionTree<ProductState, RootState> = {
...
+        })
+
+        let productIds: any = new Set();
+        products.forEach((product: any) => {
+          if(product.productId) productIds.add(product.productId);
+        })
+        productIds = [...productIds]
+        this.dispatch("product/fetchProducts", { productIds });
+
+        let variantIds: any = new Set();
```
Comment (adityasharma7) [inline src/store/modules/product/actions.ts]: I think Set conversion is not needed

File: src/store/modules/product/actions.ts [hunk 6]
```diff
@@ -83,6 +83,54 @@ const actions: ActionTree<ProductState, RootState> = {
...
+        products.forEach((product: any) => {
```
Comment (adityasharma7) [inline src/store/modules/product/actions.ts]: I think we could use reduce and map method here
```
products.reduce((acc: any, product: any) => {
          acc = [ ...acc,  product.variants.map((variant: any) => {
            variant.productId)
          })]
        }, [])
```

File: src/views/FindProductInventory.vue [hunk 7]
```diff
@@ -140,66 +140,66 @@
 
           <hr />
 
-          <div class="product" @click="() => router.push('/product-inventory')">
+          <div class="product" v-for="product in products" :key="product.groupValue" @click="() => router.push('/product-inventory')">
             <div class="desktop-only">
-              <Image src="https://cdn.shopify.com/s/files/1/0069/7384/9727/products/test-track.jpg?v=1626255137" />
+              <Image :src="getProduct(product.productId).mainImageUrl" />
```
Comment (adityasharma7) [inline src/views/FindProductInventory.vue]: I think mainImageUrl is directly available in product

File: src/views/FindProductInventory.vue [hunk 8]
```diff
@@ -140,66 +140,66 @@
...
                     <ion-thumbnail slot="start" class="mobile-only">
-                      <Image src="https://cdn.shopify.com/s/files/1/0069/7384/9727/products/test-track.jpg?v=1626255137" />
+                      <Image :src="getProduct(product.productId).mainImageUrl" />
                     </ion-thumbnail>
                     <ion-label>
-                      <p>Brand</p>
-                      Virtual Name
-                      <p>{{ $t("Color") }} : color</p>
-                      <p>{{ $t("Size") }}: size</p>
+                      <p>{{ getProduct(product.productId).brandName }}</p>
```
Comment (adityasharma7) [inline src/views/FindProductInventory.vue]: I think we should not use getProduct() getter here as product has all these information

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts [hunk 9]
```diff
@@ -83,6 +91,63 @@ const actions: ActionTree<ProductState, RootState> = {
...
+
+        let productIds: any = new Set();
+        products.forEach((product: any) => {
+          if(product.productId) productIds.add(product.productId);
+        })
+        productIds = [...productIds]
+        const productInformation = await this.dispatch("product/fetchProducts", { productIds });
+
+        products = products.map((product: any) => {
+          return {
```
Comment (adityasharma7) [inline src/store/modules/product/actions.ts:126]: Store productInformation[product.productId] in a variable and then use

File: src/views/FindProductInventory.vue [hunk 10]
```diff
@@ -140,81 +140,85 @@
 
           <hr />
 
-          <div class="product" @click="() => router.push('/product-inventory')">
+          <div class="product" v-for="product in products" :key="product.groupValue" @click="() => router.push('/product-inventory')">
```
Comment (adityasharma7) [inline src/views/FindProductInventory.vue]:
```suggestion
          <div class="product" v-for="product in products" :key="product.productId" @click="() => router.push('/product-inventory')">
```

File: src/views/FindProductInventory.vue [hunk 11]
```diff
@@ -140,81 +140,85 @@
...
 
               <div class="desktop-only">
                 <ion-list>
                   <ion-list-header>
-                    Variants
+                    {{ $t("variants") }}
                     <hr />
                   </ion-list-header>
-                  <div v-for="item in 2" :key="item" class="list-item">
+                  <div v-for="item in product.variants" :key="item.productId" class="list-item">
```
Comment (adityasharma7) [inline src/views/FindProductInventory.vue]:
```suggestion
                  <div v-for="variant in product.variants" :key="item.productId" class="list-item">
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts (hunk 9 above)
Comment (meet-aniket) [reply src/store/modules/product/actions.ts:126]: Sir, modified code as per your suggestions.

File: src/views/FindProductInventory.vue (hunk 10 above)
Comment (meet-aniket) [reply src/views/FindProductInventory.vue]: changed groupValue with productId.

File: src/views/FindProductInventory.vue (hunk 11 above)
Comment (meet-aniket) [reply src/views/FindProductInventory.vue]: Changed variable name from `item` to `variant`.

File: src/views/FindProductInventory.vue (hunk 8 above)
Comment (meet-aniket) [reply src/views/FindProductInventory.vue]: removed getter from view page.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**